cmd/titus-metadata-service/main.go

```diff
@@ -235,29 +202,44 @@ func main() {
 			EnvVar:      "X_FORWARDED_FOR_BLOCKING_MODE",
 			Destination: &xFordwardedForBlockingMode,
 		},
+		cli.StringFlag{
+			Name:        "peer-namespace",
+			Usage:       "When set, the proxy will bind inside the namespace specified",
+			EnvVar:      "PEER_NAMESPACE",
+			Destination: &peerNs,
+		},
 	}
+
 	app.Action = func(c *cli.Context) error {
 		if debug {
 			log.SetLevel(log.DebugLevel)
 		} else {
 			log.SetLevel(log.InfoLevel)
 		}
-		// This needs to be if journald available, because titus executors are started by mesos agent, and in order
-		// for logs to get from the tasks starts by titus tasks to
-		logsutil.MaybeSetupLoggerIfOnJournaldAvailable()
+
+		utils.MaybeSetupLoggerIfOnJournaldAvailable()
+
 		/* Get the requisite configuration from environment variables */
-		listener := getListener(listenPort, listenerFd)
+		var listener net.Listener
+
+		if len(peerNs) > 0 {
+			// We were launched by a CNI. Bind inside peer namespace.
+			log.Infof("Launched with PEER_NS %s, LISTEN_PORT %d", peerNs, listenPort)
+			nsListener, err := utils.GetNsListener(peerNs, listenPort)
+			if err != nil {
+				log.Fatalf("Error getting listener %s", err)
```

Comment:
Shouldn't this return an error?

so:
`return errors.Wrap(err, "Could not get listener")`

Reply:
It's the main method of the metadata proxy.